group_layers/group-layers/src/main/java/com/esri/samples/group_layers/GroupLayersSample.java

```diff
@@ -78,40 +82,63 @@ public void start(Stage stage) {
-
-      // add the group layer and other layers to the scene as operational layers
-      scene.getOperationalLayers().addAll(Arrays.asList(groupLayer, devBBuildings, devProjectArea));
-
-      // zoom to the extent of the group layer when the child layers are loaded
-      groupLayer.getLayers().forEach(childLayer ->
-        childLayer.addDoneLoadingListener(() -> {
-          if (childLayer.getLoadStatus() == LoadStatus.LOADED) {
-            sceneView.setViewpointCamera(new Camera(groupLayer.getFullExtent().getCenter(), 700, 0, 60, 0));
+      // create two group layers, set the visibility modes and add layers
+      projectAreaGroupLayer = new GroupLayer();
+      projectAreaGroupLayer.setName("Project area group");
+      projectAreaGroupLayer.setVisibilityMode(GroupVisibilityMode.INDEPENDENT);
+      projectAreaGroupLayer.getLayers().addAll(Arrays.asList(
+        (new FeatureLayer(new ServiceFeatureTable("https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevelopmentProjectArea/FeatureServer/0"))),
+        (new FeatureLayer(new ServiceFeatureTable("https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_Pathways/FeatureServer/1"))),
+        (new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_Trees/SceneServer"))
+      ));
+
+      buildingsGroupLayer = new GroupLayer();
+      buildingsGroupLayer.setName("Buildings group");
+      buildingsGroupLayer.setVisibilityMode(GroupVisibilityMode.EXCLUSIVE);
+      buildingsGroupLayer.getLayers().addAll(Arrays.asList(
+        (new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevA_BuildingShells/SceneServer")),
+        (new ArcGISSceneLayer("https://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/DevB_BuildingShells/SceneServer"))
+      ));
+
+      // load the group layers
+      projectAreaGroupLayer.loadAsync();
+      buildingsGroupLayer.loadAsync();
+
+      // Display an alert if the child layers in the group layers fail to load
+      projectAreaGroupLayer.getLayers().forEach(layer ->
+        layer.addDoneLoadingListener(() -> {
+          if (layer.getLoadStatus() == LoadStatus.LOADED) {
+            // zoom to the extent of the project area group layer when the child layers are loaded
+            sceneView.setViewpointCamera(new Camera(projectAreaGroupLayer.getFullExtent().getCenter(), 700, 0, 60, 0));
+          } else new Alert(Alert.AlertType.ERROR, "Layer failed to load:\n" + layer.getLoadError().getCause().getMessage()).show();
+        })
+      );
+
+      buildingsGroupLayer.getLayers().forEach(layer ->
+        layer.addDoneLoadingListener(() -> {
+          if (layer.getLoadStatus() != LoadStatus.LOADED) {
+            new Alert(Alert.AlertType.ERROR, "Layer failed to load:\n" + layer.getLoadError().getCause().getMessage()).show();
           }
         })
       );
```

Comment:
I think we can add the alert into `buildLayersView` where it's already looping through all the layers in the sample anyway and saves us repeating some code. If that doesn't work out, perhaps an additional method to save repeating the code? 
```suggestion
      // zoom to the extent of the group layer when the child layers are loaded
      projectAreaGroupLayer.getLayers().forEach(childLayer ->
        childLayer.addDoneLoadingListener(() -> {
          if (childLayer.getLoadStatus() == LoadStatus.LOADED) {
            // zoom to the extent of the project area group layer when the child layers are loaded
            sceneView.setViewpointCamera(new Camera(projectAreaGroupLayer.getFullExtent().getCenter(), 700, 0, 60, 0));
          }
        })
      );
```

Reply:
Added this and tested and it works well 👍